### PR TITLE
Add new cryptocurrency

### PR DESCRIPTION
**Pull request by emanuelLeiva**

### CODE REVIEW

File: App.tsx [hunk 1]
```diff
@@ -5,25 +5,38 @@ import theme from './src/themes';
...
-            options={{
-              headerShown: false,
-            }}
-          />
-        </stack.Navigator>
-      </NavigationContainer>
-    </ThemeProvider>
-  </>
+  <Provider store={store}>
+    <>
```
Comment (FedericoLeiva12) [inline App.tsx:17]: delete this <>

File: App.tsx [hunk 2]
```diff
@@ -5,25 +5,38 @@ import theme from './src/themes';
...
+              name="Laotrascreen"
+              component={AddCryptoScreen}
+              options={{
+                headerShown: false,
+              }}
+            />
+          </stack.Navigator>
+        </NavigationContainer>
+      </ThemeProvider>
+    </>
```
Comment (FedericoLeiva12) [inline App.tsx:38]: delete this

File: src/components/AddCrypto/index.tsx [hunk 3]
```diff
@@ -1,9 +1,10 @@
-import {TouchableOpacity, Text} from 'react-native';
```
Comment (FedericoLeiva12) [inline src/components/AddCrypto/index.tsx:1]: Change the name of the folder to Button

File: src/components/cryptoList/types.ts [hunk 4]
```diff
@@ -9,6 +9,7 @@ export type CryptoitemsProps = {
 };
 
 export type Crypto = {
+  data(data: any): {payload: any; type: 'crypto/add'};
```
Comment (FedericoLeiva12) [inline src/components/cryptoList/types.ts:12]: Is this data type needed?

File: src/components/cryptoList/types.ts [hunk 5]
```diff
@@ -18,3 +19,7 @@ export type Crypto = {
     percent_change_usd_last_24_hours: number;
   };
 };
+
+export type cryptoTypes = {
+  state: undefined;
```
Comment (FedericoLeiva12) [inline src/components/cryptoList/types.ts:24]: state isn't undefined, change it to the types you save in the state (cryptos)

File: src/redux/cryptoSlice.ts [hunk 6]
```diff
@@ -0,0 +1,23 @@
+import {createSlice} from '@reduxjs/toolkit';
+import {InitialStateType} from './types';
+
+const initialState: InitialStateType = {
+  cryptos: [],
+  crypto: [],
```
Comment (FedericoLeiva12) [inline src/redux/cryptoSlice.ts:6]: You aren't using this

File: src/redux/cryptoSlice.ts [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+  id: '',
```
Comment (FedericoLeiva12) [inline src/redux/cryptoSlice.ts:7]: You are not using this

File: src/redux/cryptoSlice.ts [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+  State: ''
```
Comment (FedericoLeiva12) [inline src/redux/cryptoSlice.ts:8]: You aren't using this

File: src/redux/cryptoSlice.ts [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+};
+
+const cryptoSlice = createSlice({
+  name: 'crypto',
+  initialState,
+  reducers: {
+    add: (state, {payload}) => {
+      if (state.cryptos.find(crypto => crypto.id === payload.id) === undefined)
```
Comment (FedericoLeiva12) [inline src/redux/cryptoSlice.ts:16]: use destructuring in crypto: ({ id })

File: src/screens/AddCryptoScreen/index.tsx [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+interface LaotrascreenProps {
+  navigation: {
+    navigate: (routeName: string) => void;
+  };
+  useFocused: {
+    focused: boolean;
+  };
+}
+
+const Laotrascreen = ({navigation}: LaotrascreenProps) => {
```
Comment (FedericoLeiva12) [inline src/screens/AddCryptoScreen/index.tsx:20]: Change the name of the screen

File: src/screens/AddCryptoScreen/index.tsx [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+import {Container} from '../../components/cryptoList/style';
+import {AddBut, AddCryp, AddText, TextCrypro, TitleAdd} from './style';
+import {useDispatch} from 'react-redux';
+import React, {useState} from 'react';
+import {useLazyGetCryptoQuery} from '../../redux/api/crypto';
+import {useEffect} from 'react';
+import {add} from '../../redux/cryptoSlice';
+import theme from '../../themes';
+
+interface LaotrascreenProps {
```
Comment (FedericoLeiva12) [inline src/screens/AddCryptoScreen/index.tsx:11]: Change the name of the interface

File: App.tsx [hunk 12]
```diff
@@ -5,25 +5,38 @@ import theme from './src/themes';
...
+  <Provider store={store}>
+    <>
+      <ThemeProvider theme={theme}>
+        <NavigationContainer>
+          <stack.Navigator initialRouteName="crypto">
+            <stack.Screen
+              name="crypto"
+              component={Cryptos}
+              options={{
+                headerShown: false,
```
Comment (tomasdematos) [inline App.tsx:25]: move this to the options of the navigator, instead of the options of all the screens

File: src/screens/cryptos/index.tsx [hunk 13]
```diff
@@ -31,9 +24,10 @@ function List() {
           renderItem={({item}) => <Cryptoitems item={item} />}
         />
       </View>
-      <AddButton>
-        <Button />
-      </AddButton>
+      <Button
+        text="+ Add a Cryptocurrency"
+        onPress={() => navigation.navigate('Laotrascreen')}
```
Comment (FedericoLeiva12) [inline src/screens/cryptos/index.tsx:29]: change the screen name

File: App.tsx [hunk 14]
```diff
@@ -5,25 +5,38 @@ import theme from './src/themes';
...
+              }}
+            />
+            <stack.Screen
+              name="Laotrascreen"
```
Comment (FedericoLeiva12) [inline App.tsx:29]: Change the screen name

File: src/redux/actions.ts [hunk 15]
```diff
@@ -0,0 +1,30 @@
+import {createAsyncThunk} from '@reduxjs/toolkit';
+import {API_URL} from '@env';
+import axios from 'axios';
+import {CryptoType} from './types';
+
+export const fetchCrypto = createAsyncThunk(
```
Comment (FedericoLeiva12) [inline src/redux/actions.ts:6]: you are not using this action

File: src/redux/actions.ts [hunk 16]
```diff
@@ -0,0 +1,30 @@
...
+    try {
+      const response = await axios.get(`${API_URL}${code}/metrics#`);
+      return response.data;
+    } catch (error) {
+      throw new Error('Fetch error');
+    }
+  },
+);
+
+export const updateFetchCrypto = createAsyncThunk(
```
Comment (FedericoLeiva12) [inline src/redux/actions.ts:18]: You are not using this action

File: src/redux/store.ts [hunk 17]
```diff
@@ -0,0 +1,15 @@
+import cryptoReducer from './cryptoSlice';
+import {Action, configureStore, Store} from '@reduxjs/toolkit';
+import {criptoApi} from './api/crypto';
+
+export const store: Store<{[key: string]: any}, Action<any>> = configureStore({
```
Comment (tomasdematos) [inline src/redux/store.ts:5]: don't use any

File: src/redux/types.ts [hunk 18]
```diff
@@ -0,0 +1,24 @@
+import {ImageSourcePropType} from 'react-native';
+
+export type InitialStateType = {
+  cryptos: Crypto[];
+  crypto: Crypto[];
+  id: string | undefined;
```
Comment (tomasdematos) [inline src/redux/types.ts:6]: use: 
id?: string 

? is for optional values

File: src/screens/AddCryptoScreen/index.tsx [hunk 19]
```diff
@@ -0,0 +1,73 @@
...
+      <Container>
+        <TitleAdd>
+          <Text>Add a Cryptocurrency</Text>
+        </TitleAdd>
+      </Container>
+
+      <View>
+        <TextCrypro
+          placeholder="Use a name or ticker symbol..."
+          onChangeText={text => setInputValue(text)}
```
Comment (tomasdematos) [inline src/screens/AddCryptoScreen/index.tsx:57]: use:
          onChangeText={setInputValue}

it takes the value of the text

File: src/screens/AddCryptoScreen/style.tsx [hunk 20]
```diff
@@ -0,0 +1,45 @@
...
+  border-width: 1px;
+  padding: 13px;
+  margin-top: 24px;
+  margin-left: 24px;
+  margin-right: 24px;
+  border-color: ${theme.colors.divider};
+  background-color: ${theme.colors.separator};
+`;
+export const AddCryp = styled(TouchableOpacity)`
+  position: absolute;
```
Comment (tomasdematos) [inline src/screens/AddCryptoScreen/style.tsx:30]: why position absolute, for the button? 
use flex to position it

File: src/screens/AddCryptoScreen/style.tsx [hunk 21]
```diff
@@ -0,0 +1,54 @@
+import styled from 'styled-components/native';
+import theme from '../../themes';
+import {Text, TouchableOpacity, View} from 'react-native';
+import {TextInput} from 'react-native-gesture-handler';
+export const Container = styled(View)``;
```
Comment (tomasdematos) [inline src/screens/AddCryptoScreen/style.tsx:5]: unnecessary

File: src/screens/AddCryptoScreen/style.tsx [hunk 22]
```diff
@@ -0,0 +1,54 @@
...
+
+export const AddBut = styled(Text)`
+  padding-top: 16px;
+  margin-left: 24px;
+`;
+export const BackBot = styled(Text)`
```
Comment (tomasdematos) [inline src/screens/AddCryptoScreen/style.tsx:11]: add a empty line above

Comment (tomasdematos) [reply src/screens/AddCryptoScreen/style.tsx:11]: also, BackBot? or BackBut?

File: src/screens/AddCryptoScreen/style.tsx [hunk 23]
```diff
@@ -0,0 +1,54 @@
...
+  height: 35px;
+  margin-left: 15px;
+  color: ${theme.colors.name};
+`;
+export const ContDuo = styled(View)`
+  margin-left: 25px;
+  margin-right: 25px;
+  display: flex;
+`;
+export const ContThird = styled(Container)`
```
Comment (tomasdematos) [inline src/screens/AddCryptoScreen/style.tsx:29]: Use more splicit names, in all files

File: src/components/Button/index.tsx [hunk 24]
```diff
@@ -0,0 +1,16 @@
+import {Text} from 'react-native';
+import React from 'react';
+import {AddButton} from './style';
+
+type ButtonProps = {
+  text: string;
+  onPress: () => void;
+};
+
+const Button: React.FC<ButtonProps> = ({text, onPress}) => (
```
Comment (FedericoLeiva12) [inline src/components/Button/index.tsx:10]: FC is deprecated, avoid using it

File: src/redux/api/crypto.ts [hunk 25]
```diff
@@ -0,0 +1,17 @@
+import {createApi, fetchBaseQuery} from '@reduxjs/toolkit/query/react';
+import {API_URL} from '@env';
+import {Crypto} from '../../components/cryptoList/types';
+
+type GetCryptoResponse = Crypto;
+
+export const criptoApi = createApi({
```
Comment (VicenBayugar) [inline src/redux/api/crypto.ts:7]: cryptoApi

**Pull request merged**